src/main/java/org/gridsuite/directory/server/repository/DirectoryElementRepository.java

```diff
@@ -73,17 +73,16 @@ interface SubDirectoryCount {
     @Query("SELECT e FROM DirectoryElementEntity e " +
             "WHERE e.id IN :uuids " +
             "AND e.stashed = :stashed " +
-            "AND (e.owner = :userId OR e.isPrivate = false OR (e.isPrivate IS NULL AND NOT EXISTS (SELECT 1 FROM DirectoryElementEntity parent WHERE parent.id = e.parentId AND parent.isPrivate = true)))")
+            "AND e.owner = :userId ")
     List<DirectoryElementEntity> findAllStashedElements(@Param("uuids") List<UUID> uuids,
```

Comment:
Then it's now writable by a JPA Query method :
`findByIdInAndStashedTrueAndOwner`